### DISCUSSION TITLE
support type checking in the vscode web extension

### DISCUSSION THREAD

**Discussion by DetachHead** [microsoft/pylance-release#5797] [Enhancement]
when installing pylance on https://vscode.dev, it shows the warning: "Analysis is limited to opened files."

![image](https://github.com/microsoft/pylance-release/assets/57028336/ba6d7d73-92ff-4027-b40c-4135a2b9a840)

however it seems that it's more limited than that, and that type checking is not supported at all:

![image](https://github.com/microsoft/pylance-release/assets/57028336/4d78710f-dab5-4720-a043-c6b6df1d7813)

**Comment by heejaechang** [comment]
it depends on what kind of type checking you are talking about. for type checking to be enabled.

we need to figure out how to do these 2.

1. support cancellation
2. either provide a way to install third party packages that are not part of bundled stubs or a way to disable type errors originated from third party packages that don't exist in vscode.dev

we have a way to handle (1) but not sure about (2)
